src/angular/src/app/spark-core-angular/components/sprk-icon/sprk-icon.component.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { SparkIconComponent } from './sprk-icon.component';
+
+describe('SparkAlertComponent', () => {
+  let component: SparkIconComponent;
+  let fixture: ComponentFixture<SparkIconComponent>;
+  let linkElement: HTMLElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [SparkIconComponent]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SparkIconComponent);
+    component = fixture.componentInstance;
+    linkElement = fixture.nativeElement.querySelector('button');
+  })
+
+  it('should create itself', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('getClasses should match what gets set on the element', () => {
+    fixture.detectChanges();
+    expect(linkElement.classList.toString()).toEqual(component.getClasses());
+  });
+
+  it('should add the correct class if iconType is not set', () => {
+    expect(component.getClasses()).toEqual('sprk-c-Icon');
+  });
+
+  it('should add the correct class if iconType is bell', () => {
+    component.iconType = 'bell';
+    expect(component.getClasses()).toEqual('sprk-c-Icon');
+  });
+
+  it('should add the correct classes if iconType has no value, but additionalClasses does', () => {
+    component.additionalClasses = 'sprk-u-pam sprk-u-man';
```

Comment:
are these tests passing? i thought you would need fixture.detectChanges() for setting @input data

Reply:
@RCopeland  hey rob, this is labeled with the "DO NOT MERGE" label. It's not ready for review. I know it auto-requested your review so I through that label on there.